stetho/src/main/java/com/facebook/stetho/inspector/network/AsyncPrettyPrinterExecutorHolder.java

```diff
@@ -16,5 +16,21 @@
  * A holder class for the executor service used for pretty printing related tasks
  */
 class AsyncPrettyPrinterExecutorHolder {
-  public static final ExecutorService sExecutorService = Executors.newCachedThreadPool();
+  private static ExecutorService sExecutorService;
+
+  public static void init() {
```

Comment:
The ExecutorService for async pretty printing is being used at two instances. First instance occurs when Stetho receives a network response. At that point, Stetho will try to create a pretty printer and ExecutorService will start downloading schema if pretty printer exists for that response (from DownloadingAsyncPrettyPrinterFactory's getInstance method). Second instance occurs when the user clicks on the Response tab in Network to view the response body. Then ExecutorService is used again in ResponseBodyFileManager to asynchronously pretty print response. The first instance occurs before the second instance. I can't pass the executor service from ResponeBodyFileManager to DownloadingAsyncPrettyPrinterFactory.